File: src/com/anathema_roguelike/characters/inventory/Neck.java
package com.anathema_roguelike.characters.inventory;

import java.util.Collection;
import java.util.HashSet;

import com.anathema_roguelike.characters.Character;
import com.anathema_roguelike.items.Amulet;

public class Neck extends Slot<Amulet> {
	
	HashSet<Amulet> amulets = new HashSet<>();
	
	public Neck(Character character) {
		super(Amulet.class, character);
	}

	@Override
	public Collection<Amulet> getEquippedItems() {
		return amulets;
	}

	@Override
	public void equip(Amulet item) {
		amulets.add(item);
		item.equippedTo(getCharacter());
	}

	@Override
	void remove(Amulet item) {
		if(amulets.contains(item)) {
			amulets.remove(item);
			item.removedFrom(getCharacter());
			
			getCharacter().getInventory().pickUp(item);
		}
		
	}
}

File: src/com/anathema_roguelike/entities/characters/inventory/Neck.scala


package com.anathema_roguelike
package entities.characters.inventory

import com.anathema_roguelike.entities.items.Amulet
import com.anathema_roguelike.entities.characters.Character

import scala.collection.mutable

class Neck(character: Character) extends Slot[Amulet](character) {
  private val amulets: mutable.Set[Amulet] = mutable.Set[Amulet]()

  override def getEquippedItems: Iterable[Amulet] = amulets

  def equip(item: Option[Amulet]): Unit = {
    item.foreach(i => {
      amulets.add(i)
      i.equippedTo(getCharacter)
    })
  }

  override def remove(item: Option[Amulet]): Unit = {
    item.foreach(i => {
      if (amulets.contains(i)) {
        amulets.remove(i)
        i.removedFrom(getCharacter)
        getCharacter.getInventory.pickUp(i)
      }
    })
  }
}
